src/Microsoft.ML.Data/Evaluators/Metrics/BinaryPrecisionRecallDataPoint.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// This class represents one data point on Precision-Recall curve for binary classification.
+    /// </summary>
+    public sealed class BinaryPrecisionRecallDataPoint
+    {
+        /// <summary>
+        /// Gets the threshold for this data point.
+        /// </summary>
+        public double Threshold { get; }
+        /// <summary>
+        /// Gets the precision for the current threshold.
+        /// </summary>
+        public double Precision { get; }
+        /// <summary>
+        /// Gets the recall for the current threshold.
+        /// </summary>
+        public double Recall { get; }
+
+        /// <summary>
+        /// Gets the true positive rate for the current threshold.
+        /// </summary>
+        public double TruePositiveRate => Recall;
```

Comment:
>public double TruePositiveRate => Recall; [](start = 8, length = 41)

So, what is the point of this?

Reply:
This is to make life easier for users to plot ROC curve. ROC curve is plotted as TPR vs FPR. We do have FPR, but spent some time with Rogan looking for TPR in order to get ROC data. It appears Recall is synonymous with TPR

---
In reply to: [268370748](https://github.com/dotnet/machinelearning/pull/3039#discussion_r268370748) [](ancestors = 268370748)